airbyte-integrations/connectors/source-faker/source_faker/source.py

```diff
@@ -0,0 +1,146 @@
+        catalog = read_json(spec_path)
+        streams.append(AirbyteStream(name="Users", json_schema=catalog, supported_sync_modes=["full_refresh", "incremental"]))
+        return AirbyteCatalog(streams=streams)
+
+    def read(
+        self, logger: AirbyteLogger, config: json, catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
+    ) -> Generator[AirbyteMessage, None, None]:
+        """
+        Returns a generator of the AirbyteMessages generated by reading the source with the given configuration,
+        catalog, and state.
+
+        :param logger: Logging object to display debug/info/error to the logs
+            (logs will not be accessible via airbyte UI if they are not passed to this logger)
+        :param config: Json object containing the configuration of this source, content of this json is as specified in
+            the properties of the spec.json file
+        :param catalog: The input catalog is a ConfiguredAirbyteCatalog which is almost the same as AirbyteCatalog
+            returned by discover(), but
+        in addition, it's been configured in the UI! For each particular stream and field, there may have been provided
+        with extra modifications such as: filtering streams and/or columns out, renaming some entities, etc
+        :param state: When a Airbyte reads data from a source, it might need to keep a checkpoint cursor to resume
+            replication in the future from that saved checkpoint.
+            This is the object that is provided with state from previous runs and avoid replicating the entire set of
+            data everytime.
+
+        :return: A generator that produces a stream of AirbyteRecordMessage contained in AirbyteMessage object.
+        """
+
+        count: int = config["count"] if "count" in config else 0
+        seed: int = config["seed"] if "seed" in config else state["seed"] if "seed" in state else None
+        Faker.seed(seed)
+        fake = Faker()
+
+        for stream in catalog.streams:
+            if stream.stream.name == "Users":
+                records_to_create = count
+                cursor = get_stream_cursor(state, stream.stream.name)
+                total_records = cursor
+                if stream.sync_mode == SyncMode.incremental:
+                    records_to_create = records_to_create - cursor
+
+                for i in range(0, records_to_create):
+                    yield AirbyteMessage(
+                        type=Type.RECORD,
+                        record=AirbyteRecordMessage(
+                            stream=stream.stream.name, data=emit_user(fake, i), emitted_at=int(datetime.now().timestamp()) * 1000
+                        ),
+                    )
+                    total_records = total_records + 1
+
+                yield emit_state(stream.stream.name, total_records, seed)
+            else:
+                raise ValueError(stream.stream.name)
+
+
+def get_stream_cursor(state, stream) -> int:
```

Comment:
type hints here would be good, I think `state: Dict[str, any]` and `stream: str` since it's just the stream name right?